selectrum.el

```diff
@@ -1369,42 +1296,34 @@ the update."
            (minibuf-after-string (or default " "))
            (inserted-res
             (selectrum--insert-candidates
-             selectrum-display-style
-             selectrum--refined-candidates
-             buffer
              window
              input
              ;; FIXME: This only takes our count overlay into
              ;; account there might be other overlays prefixing the
              ;; prompt.
-             (length count-info)
-             ;; Exclude selected prompt.
-             (when (and selectrum--current-candidate-index
-                        (not (< selectrum--current-candidate-index 0)))
-               selectrum--current-candidate-index)
-             (1- (length selectrum--refined-candidates))
-             selectrum--first-index-displayed
-             selectrum--actual-num-candidates-displayed))
+             (length count-info)))
            (horizp (car inserted-res))
-           (inserted-num (cdr inserted-res)))
-      (setq-local selectrum--actual-num-candidates-displayed inserted-num)
-      ;; Add padding for scrolled prompt.
-      (when (and (window-minibuffer-p window)
-                 (not horizp)
-                 (not (zerop (window-hscroll window))))
-        (let ((padding (make-string (window-hscroll window) ?\s)))
-          (with-current-buffer buffer
-            (goto-char (point-min))
-            (while (not (eobp))
-              (insert padding)
-              (forward-line 1)))))
-      (unless (or selectrum-display-action
-                  (zerop selectrum--actual-num-candidates-displayed)
-                  (not selectrum--refined-candidates))
-        (setq minibuf-after-string
-              (concat minibuf-after-string
-                      (with-current-buffer buffer
-                        (buffer-string)))))
+           (inserted-string (cadddr inserted-res)))
+      (setq-local selectrum--actual-num-candidates-displayed (cadr inserted-res))
+      (setq-local selectrum--first-index-displayed (caddr inserted-res))
+      (if selectrum-display-action
+          ;; Insert candidates into window buffer
+          (when window
+            (with-current-buffer (window-buffer window)
+              (erase-buffer)
+              (insert inserted-string)))
+        ;; Candidates are shown in minibuffer, act accordingly.
```

Comment:
Having the buffer available for minibuffer sessions is useful for debugging face issues, we could add a command instead which drops you into a buffer with the current candidate string.

Reply:
I know that you introduced it as a debugging feature, but this is no justification to pollute the display code in my opinion. My proposal would be to *not add* such a functionality. Instead you can always use a display-style in case you want to debug. I think that is good enough? As a compromise one could hide this behind a configuration option, or modify `selectrum-display-action`, to allow `'(display-buffer-some-window selectrum-show-minibuffer)`. Before passing the value to the Emacs facilities, remove `selectrum-show-minibuffer`. And only show the candidates in the minibuffer when `selectrum-show-minibuffer` is given. What do you think about that?

Reply:
I like the idea to configure minibuffer display for `selectrum-display-action`. Including a command to make debugging easy could still be nice but we haven't had this before and this functionality was kind of hidden before so I think we can deal with this at another time.

Reply:
I also like this idea. But it is a good idea to address this separately! Let's keep it in mind.